google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStoragePath.java

```diff
@@ -74,6 +74,8 @@ public String bucket() {
    * Returns path converted to a {@link BlobId} so I/O can be performed.
    */
   BlobId getBlobId() {
+    checkArgument(!path.toString().isEmpty(), 
+      "I/O not allowed on empty Google Cloud Storage object names.");
```

Comment:
I think this error message is awfully abstruse. Maybe just "object names cannot be empty"?